Scripts/Data/Operation/TransformPropertyExtractor.cs

```diff
@@ -0,0 +1,92 @@
+﻿namespace VRTK.Core.Data.Operation
+{
+    using UnityEngine;
+    using UnityEngine.Events;
+    using System;
+    using VRTK.Core.Process;
+
+    /// <summary>
+    /// Provides a basis for extracting properties from a Transform.
+    /// </summary>
+    /// <typeparam name="TElement">The element type to extract.</typeparam>
+    /// <typeparam name="TEvent">The event to emit on extraction.</typeparam>
+    public abstract class TransformPropertyExtractor<TElement, TEvent> : MonoBehaviour, IProcessable where TEvent : UnityEvent<TElement>, new()
```

Comment:
```suggestion
    public abstract class TransformPropertyExtractor<TElement, TEvent> : MonoBehaviour, IProcessable
    where TEvent : UnityEvent<TElement>, new()
```
Because we're doing it everywhere else in this PR and it looks nicer. We may want to add this to our code conventions.